Sources/Readability.swift

```diff
@@ -657,4 +690,118 @@ open class Readability {
 
 		return .none
 	}
+    
+    
+    
+    private func datePublished() -> String? {
+        
+        if let document = document {
+            
+            if  let documentString = String(data: (document.data!), encoding: String.Encoding.utf8) {
+                
+                let checkString = "type=\"application/ld+json\">"
+                if documentString.range(of:checkString) != .none {
+                    
+                    let scanner = Scanner(string:documentString)
+                    scanner .scanUpTo(checkString, into: .none)
+                    
+                    var scanned: NSString?
+                    scanner .scanString(checkString, into: .none)
+                    
+                    if scanner .scanUpTo("</script>", into: &scanned) {
+                        
+                        guard  let dict = convertToDictionary(text: scanned! as String) else {
+                            
+                            return .none
+                        }
+                        
+                        if let publishDate = dict["datePublished"] {
+                            return publishDate as? String
+                        } else if  let publishDate = dict["dateCreated"] {
+                            return publishDate as? String;
+                        }
+                        
+                    }
+
+                }
+                
+            }
+            if let dateData = extractValueUsing(document, queries: dateQueries){
+                return dateData
+            }
+            
+        }
+        return .none
+    }
+    func convertToDictionary(text: String) -> [String: Any]? {
+
+        let  textdata = stringByRemovingControlCharacters(astring: text)
+        if let data = textdata.data(using: .utf8) {
+            do {
+                return try JSONSerialization.jsonObject(with: data, options: []) as? [String: Any]
+            } catch {
+                print(error.localizedDescription)
+            }
+        }
+        return nil
+    }
+   private func stringByRemovingControlCharacters(astring: String) -> String {
+        let controlChars = NSCharacterSet.controlCharacters
```

Comment:
Is it possible to use CharacterSet struct instead of NSCharacterSet?

Reply:
yes its possible ... changed in new commit